plasmapy/analysis/swept_langmuir/floating_potential.py

```diff
@@ -0,0 +1,297 @@
+
+    **How it works**
+
+    1. The current array ``current`` is scanned for all points equal to zero and
+       point pairs that straddle :math:`I = 0`.  This forms an array of
+       "crossing-points."
+    2. The crossing-points are then grouped into "crossing-islands" in based on
+       the ``threshold`` keyword.
+
+       - A new island is formed when a successive crossing-point is more (index)
+         steps away from the previous crossing-point than allowed by
+         ``threshold``.
+       - If multiple crossing-islands are identified, then the span from the
+         first point in the first island to the last point in the last island is
+         compared to ``min_points``.  If the span is less than or equal to
+         ``min_points``, then that span is taken as one larger crossing-island
+         for the fit; otherwise, the function is incapable of identifying
+         :math:`V_f` and will return `numpy.nan` values.
+
+    3. To calculate the floating potential...
+
+       - If the crossing-island contains less points than ``min_points``, then
+         each side of the crossing-island is equally padded with the nearest
+         neighbor points until ``min_points`` is satisfied.
+       - A fit is then performed using `scipy.stats.linregress` for
+         ``fit_type="linear"`` and `scipy.optimize.curve_fit` for
+         ``fit_type="exponential"``.
+
+    Parameters
+    ----------
+
+    voltage: `numpy.ndarray`
+        1-D numpy array of monotonically ascending/descending probe biases
+        (should be in volts)
+
+    current: `numpy.ndarray`
+        1-D numpy array of probe current (should be in amperes) corresponding
+        to the :data:`voltage` array
+
+    threshold: positive, non-zero `int`
+        Max allowed index distance between crossing-points before a new
+        crossing-island is formed.  That is, if ``threshold=5`` then consecutive
+        crossing-points are considered to be in the same crossing-island if
+        they are within 5 index steps of each other. (Default: 1)
+
+    min_points: positive `int` or `float`
+        Minimum number of data points required for the fitting to be applied to.
+        See **How it works** above for additional details.  The following list
+        specifies the optional values:
+
+        - ``min_points = None`` (Default) The largest of 5 and
+          ``factor * array_size`` is taken, where ``array_size`` is the size of
+          ``voltage`` and ``factor = 0.1`` for ``fit_type = "linear"`` and
+          ``0.2`` for ``"exponential"``.
+        - ``min_points = 0`` The entire passed array is fitted.
+        - ``min_points >= 1`` Exact minimum number of points.
+        - ``0 < min_points < 0`` The minimum number of points is taken as
+          ``min_points * array_size``.
+
+    fit_type: str
+        The type of curve to be fitted to the Langmuir trace,  ``"linear"`` or
+        ``"exponential"`` (Default).  This selects which `FitFunction` class
```

Comment:
With regards to `default` vs `Default`, I generally prefer `Default` because I feel `default` is more easily lost among the rest of the parameter description.  I am also open to standardizing this across `plasmapy` and just adopting that standard.

I see the possible ambiguity with using `"exponential"`, but I feel it's not too bad here since `find_floating_potential` only uses the options `"linear"` and `"exponential"` for `fit_type` and only one version of the exponential fit functions is deployed.  I am prioritizing user convenience over explicitness here, since I don't feel understanding is lost by going from `"exponential_plus_offset"` -> `"exponential"`.